### PR TITLE
add dop event & fix event edit

### PR DESCRIPTION
**Pull request by rabuzarus**
The `Edit Event` button wasn't available anymore for the event modal. This PR should fix it.
In addition it is now possible to delete events

### CODE REVIEW

**Comment by annando** [comment]
I guess @hypolite will request some standard changes.

**Comment by rabuzarus** [comment]
Yes I think so. I tried my best for events.php but at some places I was not sure

File: include/event.php [hunk 1]
```diff
@@ -612,7 +612,14 @@ function process_events($arr) {
 			$is_first = ($d !== $last_date);
 
 			$last_date = $d;
-			$edit = ((! $rr['cid']) ? array(App::get_baseurl().'/events/event/'.$rr['id'],t('Edit event'),'','') : null);
+
+			// Show edit and drop actions only if the user is the owner of the event and the event
+			// is a real event (no bithdays)
+			if (local_user() && local_user() == $rr['uid'] && $rr['type'] == 'event') {
+				$edit = ((! $rr['cid']) ? array(App::get_baseurl().'/events/event/'.$rr['id'],t('Edit event'),'','') : null);
+				$drop = array(App::get_baseurl().'/events/drop/'.$rr['id'],t('Delete event'),'','');
```
Comment (Hypolite) [inline include/event.php]: Standards: Please add spaces after commas in the whole file.

File: mod/events.php [hunk 2]
```diff
@@ -96,7 +93,7 @@ function events_post(App $a) {
 	$action = ($event_id == '') ? 'new' : "event/" . $event_id;
 	$onerror_url = App::get_baseurl() . "/events/" . $action . "?summary=$summary&description=$desc&location=$location&start=$start_text&finish=$finish_text&adjust=$adjust&nofinish=$nofinish";
 
-	if (strcmp($finish,$start) < 0 && !$nofinish) {
+	if (strcmp($finish, $start) < 0 && !$nofinish) {
 		notice( t('Event can not end before it has started.') . EOL);
 		if (intval($_REQUEST['preview'])) {
 			echo( t('Event can not end before it has started.'));
```
Comment (Hypolite) [inline mod/events.php]: Standards: Please remove parentheses from language constructs. (also works for `include`/`require`, `return`, etc...)

File: mod/events.php [hunk 3]
```diff
@@ -105,9 +102,9 @@ function events_post(App $a) {
 		goaway($onerror_url);
 	}
 
-	if((! $summary) || (! $start)) {
+	if ((! $summary) || (! $start)) {
 		notice( t('Event title and start time are required.') . EOL);
-		if(intval($_REQUEST['preview'])) {
+		if (intval($_REQUEST['preview'])) {
 			echo( t('Event title and start time are required.'));
```
Comment (Hypolite) [inline mod/events.php]: Standards: Please remove parentheses from language constructs.

File: mod/events.php [hunk 4]
```diff
@@ -116,35 +113,33 @@ function events_post(App $a) {
...
+	if ($share) {
 		$str_group_allow   = perms2str($_POST['group_allow']);
 		$str_contact_allow = perms2str($_POST['contact_allow']);
 		$str_group_deny    = perms2str($_POST['group_deny']);
 		$str_contact_deny  = perms2str($_POST['contact_deny']);
 
 		// Undo the pseudo-contact of self, since there are real contacts now
-		if( strpos($str_contact_allow, '<' . $self . '>') !== false )
-		{
+		if ( strpos($str_contact_allow, '<' . $self . '>') !== false ) {
```
Comment (Hypolite) [inline mod/events.php]: Standards: Please remove the space inside the condition parentheses.

File: mod/events.php [hunk 5]
```diff
@@ -224,22 +220,23 @@ function events_content(App $a) {
...
 	));
 
 	$etpl = get_markup_template('event_end.tpl');
-	$a->page['end'] .= replace_macros($etpl,array(
+	$a->page['end'] .= replace_macros($etpl, array(
 		'$baseurl' => App::get_baseurl(),
 	));
 
-	$o ="";
+	$o ='';
```
Comment (Hypolite) [inline mod/events.php]: Standards: Please wrap operators with spaces.

File: mod/events.php [hunk 6]
```diff
@@ -267,9 +268,8 @@ function events_content(App $a) {
 	// The view mode part is similiar to /mod/cal.php
 	if ($mode == 'view') {
 
-
-		$thisyear = datetime_convert('UTC',date_default_timezone_get(),'now','Y');
-		$thismonth = datetime_convert('UTC',date_default_timezone_get(),'now','m');
+		$thisyear = datetime_convert('UTC', date_default_timezone_get(),'now', 'Y');
```
Comment (Hypolite) [inline mod/events.php]: Standards: Please add a space after commas in the whole file.

File: mod/events.php [hunk 7]
```diff
@@ -289,43 +289,43 @@ function events_content(App $a) {
...
 
-		$dim    = get_dim($y,$m);
-		$start  = sprintf('%d-%d-%d %d:%d:%d',$y,$m,1,0,0,0);
-		$finish = sprintf('%d-%d-%d %d:%d:%d',$y,$m,$dim,23,59,59);
+		$dim    = get_dim($y, $m);
+		$start  = sprintf('%d-%d-%d %d:%d:%d', $y, $m, 1, 0, 0, 0);
+		$finish = sprintf('%d-%d-%d %d:%d:%d', $y, $m, $dim, 23, 59, 59);
 
 
 		if ($a->argv[1] === 'json'){
```
Comment (Hypolite) [inline mod/events.php]: Standards: Please add a space before the condition bracket.

File: mod/events.php [hunk 8]
```diff
@@ -289,43 +289,43 @@ function events_content(App $a) {
...
 			if (x($_GET,'start'))	$start = $_GET['start'];
 			if (x($_GET,'end'))	$finish = $_GET['end'];
```
Comment (Hypolite) [inline mod/events.php]: Standards: Please add braces to the two previous conditions.

File: mod/events.php [hunk 9]
```diff
@@ -385,27 +381,31 @@ function events_content(App $a) {
...
+			'$week'  => t('week'),
+			'$day'   => t('day'),
+			'$list'  => t('list'),
 		));
 
-		if (x($_GET,'id')){ echo $o; killme(); }
+		if (x($_GET,'id')) {
+			echo $o;
+			killme();
+
```
Comment (Hypolite) [inline mod/events.php]: Standards: Please remove this extraneous new line.

File: mod/events.php [hunk 10]
```diff
@@ -415,16 +415,16 @@ function events_content(App $a) {
...
-		if(x($_REQUEST,'location')) $orig_event['location'] = $_REQUEST['location'];
-		if(x($_REQUEST,'start')) $orig_event['start'] = $_REQUEST['start'];
-		if(x($_REQUEST,'finish')) $orig_event['finish'] = $_REQUEST['finish'];
+		if (x($_REQUEST,'nofinish')) $orig_event['nofinish'] = $_REQUEST['nofinish'];
+		if (x($_REQUEST,'adjust')) $orig_event['adjust'] = $_REQUEST['adjust'];
+		if (x($_REQUEST,'summary')) $orig_event['summary'] = $_REQUEST['summary'];
+		if (x($_REQUEST,'description')) $orig_event['description'] = $_REQUEST['description'];
+		if (x($_REQUEST,'location')) $orig_event['location'] = $_REQUEST['location'];
+		if (x($_REQUEST,'start')) $orig_event['start'] = $_REQUEST['start'];
+		if (x($_REQUEST,'finish')) $orig_event['finish'] = $_REQUEST['finish'];
```
Comment (Hypolite) [inline mod/events.php]: I'd rather there was an `event` key in `$_REQUEST` that held all those parameters.

File: mod/events.php [hunk 11]
```diff
@@ -511,12 +516,34 @@ function events_content(App $a) {
...
+
+		// Delete only real events (no birthdays)
+		if (dbm::is_result($ev) && $ev[0]['type'] == 'event') {
+			$del = drop_item($ev[0]['itemid'], false);
+		}
+
+		if ($del == 0) {
+			notice( t('Failed to remove event' ) . EOL);
+		} else {
+			info( t('Event removed') . EOL);
```
Comment (Hypolite) [inline mod/events.php]: Standards: Please remove the space inside the parenthesis in the last two function calls.

File: mod/events.php [hunk 12]
```diff
@@ -511,12 +516,34 @@ function events_content(App $a) {
...
+		}
+
+		goaway(App::get_baseurl() . '/events');
+
```
Comment (Hypolite) [inline mod/events.php]: Standards: Please remove this extraneous line.

**Comment by annando** [comment]
For me it looks okay.

File: mod/events.php [hunk 13]
```diff
@@ -329,7 +321,7 @@ function events_content(App $a) {
 		);
 
 		// get events by id or by date
-		if (x($_GET,'id')){
+		if (x($_GET, 'id')){
```
Comment (Hypolite) [inline mod/events.php]: Standards: Can you please add a space before the brace?

File: include/event.php [hunk 14]
```diff
@@ -595,17 +595,17 @@ function process_events($arr) {
...
 
-			$j = (($rr['adjust']) ? datetime_convert('UTC',date_default_timezone_get(),$rr['start'], 'j') : datetime_convert('UTC','UTC',$rr['start'],'j'));
-			$d = (($rr['adjust']) ? datetime_convert('UTC',date_default_timezone_get(),$rr['start'], $fmt) : datetime_convert('UTC','UTC',$rr['start'],$fmt));
+			$j = (($rr['adjust']) ? datetime_convert('UTC', date_default_timezone_get(), $rr['start'], 'j') : datetime_convert('UTC', 'UTC', $rr['start'], 'j'));
+			$d = (($rr['adjust']) ? datetime_convert('UTC', date_default_timezone_get(), $rr['start'], $fmt) : datetime_convert('UTC', 'UTC', $rr['start'], $fmt));
 			$d = day_translate($d);
 
-			$start = (($rr['adjust']) ? datetime_convert('UTC',date_default_timezone_get(),$rr['start'], 'c') : datetime_convert('UTC','UTC',$rr['start'],'c'));
+			$start = (($rr['adjust']) ? datetime_convert('UTC', date_default_timezone_get(), $rr['start'], 'c') : datetime_convert('UTC', 'UTC', $rr['start'], 'c'));
 			if ($rr['nofinish']){
```
Comment (Hypolite) [inline include/event.php]: Standards: Can you please add a space before the brace?

File: mod/events.php [hunk 15]
```diff
@@ -360,33 +352,33 @@ function events_content(App $a) {
 			killme();
 		}
 
-		if (x($_GET,'id')){
+		if (x($_GET, 'id')){
```
Comment (Hypolite) [inline mod/events.php]: Standards: Can you please add a space before the brace?

File: include/event.php [hunk 16]
```diff
@@ -643,7 +643,7 @@ function process_events($arr) {
 				'is_first'=>$is_first,
 				'item'=>$rr,
 				'html'=>$html,
```
Comment (Hypolite) [inline include/event.php]: Standards: Can you please wrap `=>` with spaces in this array assignment?

File: mod/events.php [hunk 17]
```diff
@@ -232,18 +226,16 @@ function events_content(App $a) {
...
 
-
-
 	$mode = 'view';
 	$y = 0;
 	$m = 0;
-	$ignored = ((x($_REQUEST,'ignored')) ? intval($_REQUEST['ignored']) : 0);
+	$ignored = ((x($_REQUEST, 'ignored')) ? intval($_REQUEST['ignored']) : 0);
 
 	if($a->argc > 1) {
```
Comment (Hypolite) [inline mod/events.php]: Standards: Can you please add a space after the `if`?

**Comment by annando** [comment]
There is an upcoming merge conflict with my pull request https://github.com/friendica/friendica/pull/3243 which does a small change in ```mod/event.php```as well: https://github.com/friendica/friendica/pull/3243/files#diff-15694d39490095b7e349c6402264338fL63

Question is now in which order we should accept this pull request to haven't got too much merge problems. Nevertheless the file has to be changed after the merge to adopt to the new constant.

**Comment by annando** [comment]
@rabuzarus when you are resolving the merge conflicts, please use the new constant ```NULL_DATE``` instead of ```0000-00-00```.

File: include/event.php [hunk 18]
```diff
@@ -146,63 +152,81 @@ function format_event_bbcode($ev) {
...
 }
 
-
 function bbtoevent($s) {
 
 	$ev = array();
 
 	$match = '';
-	if(preg_match("/\[event\-summary\](.*?)\[\/event\-summary\]/is",$s,$match))
+	if (preg_match("/\[event\-summary\](.*?)\[\/event\-summary\]/is",$s,$match)) {
```
Comment (Hypolite) [inline include/event.php]: Standards: Can you please add a space after commas in this `preg_match` and all the subsequent ones?

**Comment by Hypolite** [comment]
@rabuzarus, do you feel you're done with both the NULL_DATE and standards?

**Comment by rabuzarus** [comment]
yes

**Pull request merged**